### ISSUE TITLE
Handlers missing tests

### ISSUE THREAD

**Issue by wasade** [biocore/labadmin#89]
- [x] access_control.py (@wasade, #86)
- [ ] access_decorators.py (@wasade, #88)
- [ ] ag_add_barcode_kit.py
- [ ] ag_consent_check.py
- [ ] ag_edit_barcode.py
- [ ] ag_edit_kit.py
- [ ] ag_edit_participant.py
- [ ] ag_get_participant_names.py 
- [ ] ag_new_barcode.py
- [ ] ag_new_kit.py
- [ ] ag_pulldown.py
- [ ] ag_results_ready.py
- [ ] ag_search.py
- [ ] ag_stats.py
- [ ] ag_third_party.py
- [ ] ag_update_geocode.py
- [ ] auth_handlers.py (@sjanssen2)
- [ ] barcode_util.py
- [ ] base.py (may not need to be directly tested)
- [ ] logged_in_index.py (@ElDeveloper)
- [ ] projects_summary.py

**Comment by sjanssen2** [comment]
I tried to write tests for ag_edit_participant.py but I failed to open the according website on my browser, even after creating an admin user. Is this class unused?

**Comment by wasade** [comment]
It's still a working page from what I can see. You can access it via search, and edit. To go direct, you need to specify an email as an argument

**Comment by mortonjt** [comment]
I noticed that ag_stats.py is already tested [here](https://github.com/biocore/labadmin/blob/master/knimin/tests/test_ag_stats.py) and ag_third_party.py is tested [here](https://github.com/biocore/labadmin/blob/master/knimin/tests/test_third_party.py).

Are these tests considered inadequate, or should they be taken off of this list

**Comment by wasade** [comment]
From a quick glance, there seems to be a bunch of stuff there. What are you
thoughts?

**Comment by mortonjt** [comment]
From what I can tell, it looks like these two modules are already tested ...

**Comment by wasade** [comment]
kk

**Comment by sjanssen2** [comment]
I redesigned tests for ag_stats.py because it was testing against hard coded numbers. Now it compares to stats obtained from database queries. It is part of my PR #111 , which fails otherwise, since it makes unrevertable changes to the DB.

**Comment by sjanssen2** [comment]
I'd like to know which lines of the code are not covered by my tests. I think this is normally possible with the coveralls page, but for this project it says: "The owner of this repo needs to re-authorize with github; their OAuth credentials are no longer valid so the file cannot be pulled from the github API.". Can we change this?

**Comment by wasade** [comment]
@sjanssen2, I clicked on one of the coverage links and it seemed to work? See [here](https://coveralls.io/builds/7976045)

**Comment by sjanssen2** [comment]
@wasade You are right for the general table, but if you click on one of the listed files https://coveralls.io/builds/7976045/source?filename=knimin%2Fhandlers%2Fag_new_kit.py you can see my problem.

**Comment by wasade** [comment]
@jairideout or @gregcaporaso, I think coveralls is acting odd here because of ownership under biocore. I have full admin here but I can't find a way to "authorize" this repo. I added myself to coveralls but I don't have the ability to twerk biocore repos. Any ideas here?

**Comment by gregcaporaso** [comment]
It looks like it's working:
https://coveralls.io/github/biocore/labadmin

Is it working ok for you now, or do you still need something? Let me know
if your permissions aren't what they need to be and we'll get you the right
access.

**Comment by wasade** [comment]
The main page is, but file detail isn't, can you try:

https://coveralls.io/builds/7976045/source?filename=knimin%2Fhandlers%2Fag_new_kit.py

**Comment by sjanssen2** [comment]
Do we know how a file must look like for http://localhost:8889/ag_third_party/data/ ?

**Comment by wasade** [comment]
@sjanssen2, I really don't know. @jwdebelius, do you know by chance (I believe this is raw vioscreen data)?

**Comment by gregcaporaso** [comment]
Some of the file detail pages seem to be working:
https://coveralls.io/builds/7977612/source?filename=knimin%2Flib%2Fdata_access.py

I see that the one you linked above doesn't have file detail information. If you can see the file detail on this page I doubt this is a permission issue, but let me know if there is anything you want me to try.

**Comment by wasade** [comment]
That one doesn't work for me... see the attached, or is that what you get as well?
<img width="1722" alt="screen shot 2016-09-21 at 9 25 13 am" src="https://cloud.githubusercontent.com/assets/474290/18719704/527db6c0-7fdd-11e6-9536-439b3b64589a.png">

**Comment by wasade** [comment]
Thanks @gregcaporaso btw

**Comment by gregcaporaso** [comment]
No, I see the following:

![screenshot 2016-09-21 09 26 13](https://cloud.githubusercontent.com/assets/192372/18719764/8f253a76-7fdd-11e6-91cb-b726a268b20b.png)

@jairideout is going to change your credentials - I don't think I have access to do that either (or I just don't know where to go to do it). Give us a few mins.

**Comment by jairideout** [comment]
@wasade, are you signed into the Coveralls website? I couldn't view those files until I logged in with my GitHub account.

**Comment by wasade** [comment]
Great, thank you!

**Comment by wasade** [comment]
When I sign in, I get a notice that the source isn't available...

**Comment by jairideout** [comment]
Same notice as before, or different message?

**Comment by wasade** [comment]
Different

**Comment by jairideout** [comment]
Can you see the source here?

https://coveralls.io/builds/7977612/source?filename=knimin%2Fhandlers%2Fag_results_ready.py

**Comment by wasade** [comment]
yes, @sjanssen2, are you able too as well?

**Comment by jairideout** [comment]
Okay, it wasn't a permissions issue (we have the same biocore org permissions). On one of the pages stating the source couldn't be found, there were two text fields that can be configured to help Coveralls find the files based on where they are in the Travis-CI build. The first text field specifies the absolute path to the cloned repo in Travis, and the second specifies a relative path to where the source files are in the repo. I can't remember exactly what I entered for the first text field. I deleted the contents in the second text field. Sorry I can't provide more details; now that the source files are working, Coveralls won't let me navigate to that page with the text fields. If this happens in the future, mess with those text fields until it starts working again. I just fixed the scikit-bio Coveralls source files in a similar manner.

**Comment by wasade** [comment]
Okay, very weird. Thank you for the investigation and mucking with it :) I
wonder if there was just some hiccup somewhere. It might also be the case
that people don't often check source coverage on individual files regularly
so it might be the case that this has persisted for a bit on a few projects
but just now got noticed

**Comment by gregcaporaso** [comment]
Glad it's working, thanks @jairideout!

**Comment by jairideout** [comment]
No problem! I suspect it's been an issue for awhile and ppl just don't check the individual files very often. I had to change the text fields several times until it started working, so unlikely to be a hiccup (and a similar fix to scikit-bio resolved the issue). Glad it's working now!

**Comment by sjanssen2** [comment]
Now that I am logged in to coverall, I can see the individual file coverage. Thanks for fixing it!

**Comment by sjanssen2** [comment]
@mortonjt found that there are already tests for test_ag_stats.py and test_third_party.py. I used test_ag_stats.py and made it more dynamic in an earlier PR. I updated and renamed test_third_party.py in the outstanding PR #118
